docs/markdown/snippets/feature_if_methods.md

```diff
@@ -0,0 +1,77 @@
+## Add a FeatureOption.enable_if and .disable_if
+
+These are useful when features need to be constrained to pass to `dependency()`,
+as the behavior of an `auto` and `disabled` or `enabled` feature is markably
+different. consider the following case:
+
+```meson
+feat = get_option('feature').disable_auto_if(not foo)
```

Comment:
What does "feat" even mean? :eyes: Is that like an "opt"?

Reply:
`feat` as a shortening of `feature` like `str` is a shortening of `string`? If it's not clear I can change it.

Reply:
It's not clear to me, but maybe I'm weird.